tree.go

```diff
@@ -219,23 +216,6 @@ func NewLeafNode(stem []byte, values [][]byte) *LeafNode {
 		c2:     Generator(),
 	}
 
-	// Initialize the commitment with the extension tree
```

Comment:
No, because `commitment` is `nil` in this case. In that case, the `Commit()` function already sees that as a signal that there's no previous commitment that we can do diff-updating; so it will do the usual full computation in the vector.

That means we don't need to initialize `cow` here. `cow` is initialized whenever we have a previous commitment calculation, and any value of the leaf is changed. In that case, we track in `cow` since will do a diff-update in the next `Commit()`.